File: server/sockets/packets/entity_moved.py
from server.sockets.packets import PacketType
from server.sockets.packets.packet import Packet


class EntityMovedPacket(Packet):
    def __init__(self, x: float, y: float):
        self.x = x
        self.y = y

    def dictify(self):
        return dict(type=PacketType.EntityMoved, x=self.x, y=self.y)

File: client/src/ts/packets/entitymoved.ts
import Packet, {PacketType} from './packet';

export class EntityMovedPacket extends Packet {
    public x: number;
    public y: number;

    constructor(packetDict: any) {
        super();

        this.x = packetDict['x'];
        this.y = packetDict['y'];
    }

    public dictify(): string {
        throw 42;
    }

    public getType(): PacketType {
        return PacketType.EntityMoved;
    }
}
